src/wallet/rpcwallet.cpp

```diff
@@ -110,6 +110,8 @@ UniValue getinfo(const UniValue& params, bool fHelp)
     obj.pushKV("protocolversion", PROTOCOL_VERSION);
     obj.pushKV("walletversion", pwalletMain->GetVersion());
     obj.pushKV("balance",       ValueFromAmount(pwalletMain->GetBalance()));
+    obj.pushKV("unconfirmed",   ValueFromAmount(pwalletMain->GetUnconfirmedBalance()));
+    obj.pushKV("immature",      ValueFromAmount(pwalletMain->GetImmatureBalance()));
```

Comment:
`CWallet::GetImmatureBalance()` is useless in Gridcoin because it counts coinbase transactions only. The Gridcoin version is `GetStake()`, and `GetNewMint()` does exactly the same thing. I've been ignoring these since we're going to wipe the wallet code after the next release, but maybe it's worth cleaning that up since we're looking at it.

Reply:
It is... might as well leave the "retail" calls in here, because we need to fix the underlying wallet issue.

Reply:
It is trivial to pull out the pcoin.IsCoinBase() out of GetImmatureBalance. Weird as to why that was in there in the first place...

Reply:
Actually, all three should be the same and limited to the coinstake transaction...

Reply:
I still think we should report the "immature" field and just have it be the same number as the other two. The confusing and bewildering array of terminology leads exactly to the type of questions that were posed on Discord. We can only report one field, but we will forever be answering questions about equivalency.

Reply:
`GetImmatureBalance()` was retained by Peercoin which uses both PoW and PoS so their coinbase blocks have meaningful outputs. We don't need `GetImmatureBalance()` except perhaps for extremely faithful representation by the client of the chain for the first 2050 blocks with PoW, but I really doubt that matters to anyone, and keeping something that displays that information will just add to confusion. I think the `stake` field best represents our terminology (maybe `staked` is better) and we should remove `immature` and `newmint`.

Reply:
These calls are also rather expensive for wallets with a significant transaction history, so trimming the unnecessary queries will reduce overhead and lock contention.

Reply:
Let's just have balance, unconfirmed, and stake, which is the same as the UI.